### PR TITLE
BUGZ-92: Refresh rate Throttling refinements

### PR DESCRIPTION
**Pull request by samcake**
@danteruiz Did all the work, i just took over his branch to do minor tweaks.

We are introducing a detection of the user input events to switch the regime of the refresh rate manager between "FocusActive" or FocusInactive"
On Any user input, the regime is reset to  "FocusActive" with the higher refresh rate target (Interactive: 30Hz/ Eco: 20Hz)
After the last user input event, after a wait of 3sec, the regimes goes to "FocusInactive" with the lower refresh rate target (INteractive: 20Hz / Eco: 10Hz)
If the Profile is "Realtime" then both "FocusActive" and "FocusInactive" are set at 60Hz

If in VR UX mode then the PRofile is always Realtime with a target refreshrate of 90Hz

## TEST PLAN
First that the settings/graphics indicate the "Interactive" value for refresh profile

In stats, the REfresh line (left column, 3rd line) should show a the behavior
the regime should indicate "FocusActive - 30" while interacting with key pressed or mouse moves
then after doing nothing for 3 sec it should say "FocusInactive - 20"
on the first user input it should return to "FocusActive - 30"
Clicking out of the main window  should set the indication to "Unfocus - 10"

Now  go change the settings in "graphics/Refresh" to the Eco mode
The same sequence above should give the same regime changes but different values:
"FocusActive - 20"
"FocusInactive - 10"
"Unfocus- 5"

Next change the settings in "graphics/Refresh" to the Realtime mode
The same sequence above should give the same regime changes but different values:
"FocusActive - 60"
"FocusInactive - 60"   <<<<< YES Inactive and Active are the same in Realtime profile!
"Unfocus- 10"

Lastly try the same scenarii with HMD on and then it Should show the same values regadless :
"FocusActive - 90"
"FocusInactive - 90"
"Unfocus- 10"

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19162/HighFidelity-Beta-PR15499-531ec14.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19162/HighFidelity-Beta-Interface-PR15499-531ec14.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19162/HighFidelity-Beta-PR15499-531ec14.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19162/HighFidelity-Quest-Beta-PR15499-531ec14.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19164/HighFidelity-Beta-PR15499-7f22ac1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19164/HighFidelity-Beta-Interface-PR15499-7f22ac1.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19164/HighFidelity-Beta-PR15499-7f22ac1.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19164/HighFidelity-Quest-Beta-PR15499-7f22ac1.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19164/HighFidelity-Beta-PR15499-7f22ac1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19164/HighFidelity-Beta-Interface-PR15499-7f22ac1.exe

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -4109,12 +4109,7 @@ bool Application::eventFilter(QObject* object, QEvent* event) {
     auto eventType = event->type();
     if (eventType == QEvent::KeyPress || eventType == QEvent::KeyRelease || eventType == QEvent::MouseMove) {
         RefreshRateManager& refreshRateManager = getRefreshRateManager();
-        auto refreshRateRegime = refreshRateManager.getRefreshRateRegime();
-
-        if (refreshRateRegime == RefreshRateManager::RefreshRateRegime::RUNNING ||
```
Comment (danteruiz) [inline interface/src/Application.cpp]: Can you add these lines back or these checks in the resetInactiveTimer

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19178/HighFidelity-Beta-PR15499-02530bd.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19178/HighFidelity-Quest-Beta-PR15499-02530bd.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19178/HighFidelity-Beta-PR15499-02530bd.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19178/HighFidelity-Beta-Interface-PR15499-02530bd.exe

**Comment by samcake** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19180/HighFidelity-Beta-PR15499-02530bd.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19180/HighFidelity-Beta-Interface-PR15499-02530bd.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19180/HighFidelity-Beta-PR15499-02530bd.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19180/HighFidelity-Quest-Beta-PR15499-02530bd.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19180/HighFidelity-Beta-PR15499-02530bd.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19180/HighFidelity-Beta-Interface-PR15499-02530bd.exe

**Pull request merged**